inference-engine/src/transformations/src/transformations/convert_opset1_to_legacy/convert_proposal_to_proposal_ie.cpp

```diff
@@ -8,58 +8,78 @@
 #include <vector>
 
 #include <ngraph/opsets/opset1.hpp>
-
+#include <ngraph/opsets/opset4.hpp>
+#include <ngraph/pattern/op/wrap_type.hpp>
 #include <ngraph_ops/proposal_ie.hpp>
 #include <ngraph/rt_info.hpp>
 
-ngraph::pass::ConvertProposalToLegacyMatcher::ConvertProposalToLegacyMatcher() {
-    auto input_0 = std::make_shared<pattern::op::Label>(element::f32, Shape{1, 1, 1, 1});
-    auto input_1 = std::make_shared<pattern::op::Label>(element::f32, Shape{1, 1, 1, 1});
-    auto input_2 = std::make_shared<pattern::op::Label>(element::f32, Shape{3});
+bool convert_to_proposal_ie(std::shared_ptr<ngraph::op::v0::Proposal> proposal, bool add_second_out = false) {
+    ngraph::Output<ngraph::Node> last;
 
-    ngraph::op::ProposalAttrs attr = {};
+    ngraph::NodeVector ops_to_replace, new_ops;
+    ops_to_replace.push_back(proposal);
 
-    auto proposal = std::make_shared<ngraph::opset1::Proposal>(input_0, input_1, input_2, attr);
+    if (auto reshape = std::dynamic_pointer_cast<ngraph::opset1::Reshape>(
+            proposal->input_value(2).get_node_shared_ptr())) {
+        auto input_shape = reshape->get_input_shape(0);
+        if (input_shape.size() == 2) {
+            last = reshape->input_value(0);
+            ops_to_replace.push_back(reshape);
+        }
+    }
+
+    if (!last.get_node_shared_ptr()) {
```

Comment:
This statement must always return true as Output always belongs to particular node

Reply:
Rewritten this if statement. `last` is set explicitly either to reshape node or node from where `im_info` is coming with acceptable shape. Please review